File: src/main/java/com/github/repomeisteractions/FizzBuzz.java
package com.github.repomeisteractions;

public class FizzBuzz {

    public boolean isFizzNumber(int number) {
        return number % 3 == 0;
    }

    public boolean isBuzzNumber(int number) {
        return number % 5 == 0;
    }

    public String generate(int count) {
        StringBuilder builder = new StringBuilder();
        for (int i = 1; i <= count; i++) {
            if (isFizzNumber(i) && isBuzzNumber(i))
                builder.append("FizzBuzz\n");
            else if (isFizzNumber(i))
                builder.append("Fizz\n");
            else if (isBuzzNumber(i))
                builder.append("Buzz\n");
            else
                builder.append(String.format("%d\n", i));
        }
        return builder.toString();
    }

    public static void main(String[] args) {
        FizzBuzz fizzBuzz = new FizzBuzz();
        String result = fizzBuzz.generate(100);
        System.out.println(result);
    }
}

File: fizzbuzz.js
function isFizzNumber(n) {
    return n % 3 === 0;
}

function isBuzzNumber(n) {
    return n % 5 === 0;
}

function generate(count) {
    let result = '';
    for (let i = 1; i <= count; i++) {
        if (isFizzNumber(i) && isBuzzNumber(i)) {
            result += 'FizzBuzz\n';
        } else if (isFizzNumber(i)) {
            result += 'Fizz\n';
        } else if (isBuzzNumber(i)) {
            result += 'Buzz\n';
        } else {
            result += `${i}\n`;
        }
    }
    return result;
}

module.exports = {
    isFizzNumber: isFizzNumber,
    isBuzzNumber: isBuzzNumber,
    generate: generate,
}